docs/reference/search/search-your-data/retrievers_examples.asciidoc

```diff
@@ -0,0 +1,327 @@
+tag::basic-rrf-retriever-with-semantic-query[]
+[discrete]
+=== Combining kNN and semantic search with RRF
+
+First, let's examine how to combine two different types of queries: a `kNN` query and a
+`semantic` query. While these queries may produce scores in different ranges, we can use
+Reciprocal Rank Fusion (`rrf`) to combine the results and generate a merged final result
+list.
+
+To implement this in the retriever framework, we start with the top-level element: our `rrf`
+retriever. This retriever operates on top of two other retrievers: a `knn` retriever and a
+`standard` retriever. Our query structure would look like this:
+
+[source,js]
+----
+GET /retrievers_example/_search
+{
+   "retriever":{
+       "rrf": {
+           "retrievers":[
+               {
+                   "standard":{
+                       "query":{
+                           "semantic":{
+                               "field": "inference_field",
+                               "query": "Can I use generative AI to identify user intent and improve search relevance?"
+                           }
+                       }
+                   }
+               },
+               {
+                   "knn": {
+                       "field": "vector",
+                       "query_vector": [0.23, 0.67, 0.89],
+                       "k": 3,
+                       "num_candidates": 5
+                   }
+               }
+           ],
+           "rank_window_size": 10,
+           "rank_constant": 1
+       }
+   },
+   "_source": ["text", "topic"]
+}
+----
+//NOTCONSOLE
+end::basic-rrf-retriever-with-semantic-query[]
+
+tag::rrf-retriever-with-collapse[]
+[discrete]
+=== Grouping results by year with `collapse`
+
+In our result set, we have many documents with the same `year` value. We can clean this
+up using the `collapse` parameter with our retriever. This enables grouping results by
+any field and returns only the highest-scoring document from each group. In this example
+we'll group by year.
+on a given value and pick just the top for each sub-group!
```

Comment:
Is this a leftover?

```suggestion
```

Reply:
Yeap, due to incorrect merging. Reworked the sentence to 

```
In this example we'll collapse our results based on the `year` field.
```